java/gradle.java/test/unit/src/org/netbeans/modules/gradle/java/execute/JavaExecTokenProviderTest.java

```diff
@@ -0,0 +1,248 @@
+    }
+    
+    public void testExplicitJVMArgsWithSpace() throws Exception {
+        JavaExecTokenProvider jetp = new JavaExecTokenProvider(createSimpleJavaProject());
+        
+        ExplicitProcessParameters params = ExplicitProcessParameters.builder().
+                launcherArgs(THREE_PROPS).
+                build();
+        
+        Map<String, String> map = jetp.createReplacements(ActionProvider.COMMAND_RUN, Lookups.singleton(params));
+        assertEquals(4, map.size());
+
+        assertParamList(map.get("java.jvmArgs"), THREE_PROPS);
+        assertListInValue(map.get("javaExec.jvmArgs"), THREE_PROPS);
+    }
+    
+    public void testAppParams() throws Exception {
+        JavaExecTokenProvider jetp = new JavaExecTokenProvider(createSimpleJavaProject());
+        
+        ExplicitProcessParameters params = ExplicitProcessParameters.builder().
+                args(TWO_PROPS).
+                build();
+
+        Map<String, String> map = jetp.createReplacements(ActionProvider.COMMAND_RUN, Lookups.singleton(params));
+        assertEquals(4, map.size());
+
+        assertArgs(map.get("javaExec.args"), TWO_PROPS);
+        assertParamList(map.get("java.args"), TWO_PROPS);
+    }
+    
+    public void testAppParamsWithSpace() throws Exception {
+        JavaExecTokenProvider jetp = new JavaExecTokenProvider(createSimpleJavaProject());
+        
+        ExplicitProcessParameters params = ExplicitProcessParameters.builder().
+                args(THREE_PROPS).
+                build();
+
+        Map<String, String> map = jetp.createReplacements(ActionProvider.COMMAND_RUN, Lookups.singleton(params));
+        assertEquals(4, map.size());
+
+        assertArgs(map.get("javaExec.args"), THREE_PROPS);
+        assertParamList(map.get("java.args"), THREE_PROPS);
+    }
+    
+    
+    /**
+     * Note: this code is not actually run, as it would require an entire
+     * ProjectConnection to be mocked so that the Gradle process is captured
+     * directly at execution. But the configuration code is actually tested in
+     * {@link #testExamplePassJVmAndArguments} below. This method just gives
+     * an example for javadoc, keep it synchronized with the real test !
+     *
+     */
+    public void xtestExamplePassJVmAndArguments() throws Exception {
+        Project project = createSimpleJavaProject();
+        // BEGIN: JavaExecTokenProviderTest#testExamplePassJvmAndArguments
+        ExplicitProcessParameters params = ExplicitProcessParameters.builder().
+                args("Frodo").
+                launcherArg("-verbose").
+                build();
+        ExplicitProcessParameters params2 = ExplicitProcessParameters.builder().
+                args("Baggins").replaceArgs(false).
+                launcherArg("-Dcompanion=Sam").
+                build();
+        
+        ActionProvider actions = project.getLookup().lookup(ActionProvider.class);
+        if (actions == null) {
+            // fail, no action available
```

Comment:
assert actions != null : "No actions available";

Reply:
Fixed in 5d055fdd54